site/en/docs/extensions/upgrading/api-calls/index.md

```diff
@@ -0,0 +1,206 @@
+
+Where your Manifest V2 used the `browserAction` and `pageAction` APIs, you should now use the `action` API.
+
+<div class="switcher">
+{% Compare 'worse', 'Manifest V2' %}
+```js
+chrome.browserAction.onClicked.addListener(tab => { ... });
+chrome.pageAction.onClicked.addListener(tab => { ... });
+```
+
+{% endCompare %}
+
+{% Compare 'better', 'Manifest V3' %}
+```js
+chrome.action.onClicked.addListener(tab => { ... });
+```
+
+{% endCompare %}
+</div>
+
+## Replace functions that expect a Manifest V2 background context {: #replace-mv2-function }
+
+Other extension contexts can only interact with extension service workers using [message passing](/docs/extensions/mv3/messaging/). Consequently, you'll need to replace calls that expect a background context, specifically:
+
+* `chrome.runtime.getBackgroundPage()`
+* `chrome.extension.getBackgroundPage()`
+* `chrome.extension.getExtensionTabs()`
+
+Your extension scripts should use message passing to communicate between a service worker and other parts of your extension. Currently that entails using `sendMessage()` and implementing `chrome.runtime.onMessage` in your extension service worker. Long term, you should plan to replace these calls with [`postMessage()`](https://developer.mozilla.org//docs/Web/API/Client/postMessage) and a service worker's [message event handler](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/message_event).
+
+## Replace unsupported APIs {: #replace-unsupported-apis }
+
+The methods and properties listed below need to change in Manifest V3. 
+
+| Manifest V2 method or property        | Replace with                                         |
+|---------------------------------------|------------------------------------------------------|
+| `chrome.extension.connect()`          | `chrome.extension.connect()`                         |
+| `chrome.extension.getExtensionTabs()` | `chrome.extension.getViews()`                        |
+| `chrome.extension.getURL()`           | `chrome.runtime.getURL()`                            |
+| `chrome.extension.lastError`          | Where methods return promises, use `promise.catch()` |
+| `chrome.extension.onConnect`          | `chrome.runtime.onConnect`                           |
+| `chrome.extension.onMessage`          | `chrome.runtime.onMessage`                           |
+| `chrome.extension.onRequest`          | `chrome.runtime.onRequest`                           |
+| `chrome.extension.onRequestExternal`  | `chrome.runtime.onMessageExternal`                   |
+| `chrome.extension.sendMessage()`      | `chrome.runtime.sendMessage()`                       |
+| `chrome.extension.sendRequest()`      | `chrome.runtime.sendMessage()`                       |
+| `chrome.runtime.onSuspend` (background scripts) | Not supported in extension service workers. Use the [`beforeunload`](https://developer.mozilla.org/docs/Web/API/Window/beforeunload_event) document event instead. |
```

Comment:
I rephrased and added a link. MDN [recommends agains `unload`](https://developer.mozilla.org/en-US/docs/Web/API/Window/unload_event#usage_notes) so I took it out.